TACC/Core-CMS-Resources
# refactor!: rename neuronex_cms to _3dem_cms

## Overview

Allow deploy of 3Dem on CMS v3.12:
- Rename `neuronex-cms` to `_3dem_cms`.[^1]
- Support deprecated template.

Also, add guide templates that load custom assets (to fix guide templates styles too new for the site).

> **Important**
> **After deploy to production**, to fix Guide pages missing styles, change templates:
> - from "CORE-CMS Guide: […]"
> - to "Guide: […]"

## Changes

- **changed** project name from `neuronex-cms` to `_3dem_cms`[^1]
- **added** old template path redirect, setting, and doc
- **fixed** lack of guide templates that load custom assets

## Testing & UI

1. Verify website has same styles as prod.
2. ➡️ Verify guide pages load.
3. Update guide pages to use new custom asset templates.
4. Verify guide pages have same styles as prod.
    | prod. | staging |
    | - | - |
	| <img width="990" alt="globus - prod" src="https://github.com/TACC/Core-CMS-Resources/assets/62723358/a8fcbfe5-cb36-44c9-a546-dd87f947db48"> | <img width="990" alt="globus - staging" src="https://github.com/TACC/Core-CMS-Resources/assets/62723358/30e385fc-4f9b-42bc-9ad4-cfe16300c03f"> |
	| <img width="990" alt="data - prod" src="https://github.com/TACC/Core-CMS-Resources/assets/62723358/ba1aa290-edaf-4cdc-b05a-1edd324fcc09"> | <img width="990" alt="data -staging" src="https://github.com/TACC/Core-CMS-Resources/assets/62723358/72c03884-6243-47db-8720-dd0635265296"> |
	| <img width="990" alt="start - prod" src="https://github.com/TACC/Core-CMS-Resources/assets/62723358/505e226b-88b3-450d-b1f7-2a3ea2b2eb7a"> | <img width="990" alt="start - staging - v2" src="https://github.com/TACC/Core-CMS-Resources/assets/62723358/687c3a40-f8b8-48f0-bbf2-c43eb8176426"> |
5. Verify "Getting Started (TAM)" template is available and working.
	<img width="990" alt="start - staging - tam" src="https://github.com/TACC/Core-CMS-Resources/assets/62723358/0dea7a97-33fe-4f76-9702-eb8cd3b3ec09">
6. Verify "Fullwidth" template works.
7. Verify "Fullwidth" template is clearly marked as "DEPRECATED".
    ![deprecated template](https://github.com/TACC/Core-CMS-Resources/assets/62723358/9a13e35d-5ca0-4163-a4fb-87fbfdd76390)

[^1]: The name `_3dem_cms` is prefixed with a `_` because Python will not let an app name start with a number (i.e. `3`).
